encoding/go.mod

```diff
@@ -1,3 +1,5 @@
 module github.com/philippgille/gokv/encoding
 
 go 1.18
+
+require google.golang.org/protobuf v1.31.0
```

Comment:
With the first dependencies appearing, maybe the different codecs should live in subdirectories as their own modules? Otherwise people who just use the JSON codec will still have the protobuf dependency as (unused) transitive dependency in their `go.mod`.

Reply:
PS: We'll inevitably have other codecs with dependencies anyway, so if we agree on putting each codec in a subdirectory as independent modules, it doesn't have to happen in this PR. We can merge first, maybe also the [TOML](https://github.com/philippgille/gokv/pull/103) and [MessagePack](https://github.com/philippgille/gokv/pull/98) PRs, and before releasing the next version we can do this structural change in a separate PR. But it would be a breaking change. WDYT?

Reply:
yes, make sense